### PR TITLE
JAMES-2439 Allow to disable unsecured JMX server with Guice

### PR DESCRIPTION
**Pull request by chibenwa**
This should be allowed via the configuration.

This would allow a full answer to `CVE-2017-12628`.

### CODE REVIEW

File: server/container/guice/jmx/src/main/java/org/apache/james/modules/server/JmxConfiguration.java [hunk 1]
```diff
@@ -20,35 +20,48 @@
...
+    private final Optional<Host> host;
 
     @VisibleForTesting
-    JmxConfiguration(Host host) {
+    JmxConfiguration(Optional<Host> host) {
         this.host = host;
     }
 
+    public boolean isEnabled() {
+        return host.isPresent();
```
Comment (rouazana) [inline server/container/guice/jmx/src/main/java/org/apache/james/modules/server/JmxConfiguration.java]: this is strange. I'd prefer to really have a boolean in the configuration. Optional for host could have meant configured / not configured (even if for now a default configuration is loaded).

File: src/site/xdoc/server/config-system.xml [hunk 2]
```diff
@@ -117,6 +117,8 @@
                 <p>This is used to configure the JMX MBean server via which all management is achieved (also used by via the james-cli).</p>
 
                 <dl>
+                    <dt><strong>jmx.enabled</strong></dt>
+                    <dd>(Guice only). Boolean. Should the JMX server de enabled? Defaults to `true`.</dd>
```
Comment (rouazana) [inline src/site/xdoc/server/config-system.xml]: s/de enabled/be enabled/

File: server/container/guice/jmx/src/test/java/org/apache/james/modules/server/JmxConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import java.io.StringReader;
+import java.util.Optional;
+
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.james.util.Host;
+import org.junit.jupiter.api.Test;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+public class JmxConfigurationTest {
```
Comment (blackheaven) [inline server/container/guice/jmx/src/test/java/org/apache/james/modules/server/JmxConfigurationTest.java]: Add a test to check that it's implicitly activated.

Comment (chibenwa) [reply server/container/guice/jmx/src/test/java/org/apache/james/modules/server/JmxConfigurationTest.java]: fromPropertiesShouldReturnConfiguredValues && fromPropertiesShouldReturnDefaultWhenEmpty are already testing this  (equality with an activated script).

What test exactly do you expect?

Comment (blackheaven) [reply server/container/guice/jmx/src/test/java/org/apache/james/modules/server/JmxConfigurationTest.java]: My fault, the first one does the trick, sorry

**Comment by chibenwa** [comment]
test this please

```
 no known node for /home/jenkins/build/workspace/996
```

Was built on yoda

**Comment by chibenwa** [comment]
```
[483630e3592c1a3ed5a6eff8c411053dcf187d53] [INFO] BUILD SUCCESS
[483630e3592c1a3ed5a6eff8c411053dcf187d53] [INFO] ------------------------------------------------------------------------
[483630e3592c1a3ed5a6eff8c411053dcf187d53] [INFO] Total time: 03:03 h
[483630e3592c1a3ed5a6eff8c411053dcf187d53] [INFO] Finished at: 2018-06-26T08:55:24Z
[483630e3592c1a3ed5a6eff8c411053dcf187d53] [INFO] ------------------------------------------------------------------------
[483630e3592c1a3ed5a6eff8c411053dcf187d53] Copying cassandra JARs
[483630e3592c1a3ed5a6eff8c411053dcf187d53] Copying JPA jars
[483630e3592c1a3ed5a6eff8c411053dcf187d53] Copying SPRING jars
[483630e3592c1a3ed5a6eff8c411053dcf187d53] Running: Shell Script
[483630e3592c1a3ed5a6eff8c411053dcf187d53] Running: Shell Script
[483630e3592c1a3ed5a6eff8c411053dcf187d53] Running: Print Message
[483630e3592c1a3ed5a6eff8c411053dcf187d53] Error executing docker rm --force=true --volumes=true git-publish-james-2439-93483fff-6c1d-4feb-a674-1832fb47aa74 : no known node for /home/jenkins/build/workspace/1002
```

mvn tests are OK

Can be considered :green_apple:  IMO

But test this please anyway

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**